### PR TITLE
dts: entropy: cleanup Kconfig and replace CONFIG_ENTROPY_NAME with DT_ENTROPY_NAME

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]
LGTM

**Review by dcpleung — APPROVED** [review]

File: soc/arm/atmel_sam/same70/dts_fixup.h [hunk 1]
```diff
@@ -127,7 +127,7 @@
 #define DT_USBHS_PERIPHERAL_ID	DT_ATMEL_SAM_USBHS_40038000_PERIPHERAL_ID
 #define DT_USBHS_RAM_BASE_ADDRESS	DT_ATMEL_SAM_USBHS_40038000_RAM_0
 
-#define CONFIG_ENTROPY_NAME			DT_ATMEL_SAM_TRNG_40070000_LABEL
+#define DT_ENTROPY_NAME			DT_ATMEL_SAM_TRNG_40070000_LABEL
```
Comment (MaureenHelm) [inline soc/arm/atmel_sam/same70/dts_fixup.h:130]: Why isn't this `DT_LABEL(DT_INST(0, atmel_sam_trng))`?

File: soc/arm/nxp_imx/rt/dts_fixup.h [hunk 2]
```diff
@@ -25,7 +25,7 @@
 #define DT_IRQ_ETH1_COMMON			DT_NXP_KINETIS_ETHERNET_402D4000_IRQ_0
 #define DT_IRQ_ETH1_IEEE1588_TMR		DT_NXP_KINETIS_PTP_402D4000_PTP_IRQ_0
 
-#define CONFIG_ENTROPY_NAME			DT_NXP_KINETIS_TRNG_400CC000_LABEL
+#define DT_ENTROPY_NAME			DT_NXP_KINETIS_TRNG_400CC000_LABEL
```
Comment (MaureenHelm) [inline soc/arm/nxp_imx/rt/dts_fixup.h:28]: Why isn't this `DT_LABEL(DT_INST(0, nxp_kinetis_trng))`

File: soc/arm/atmel_sam/same70/dts_fixup.h (hunk 1 above)
Comment (galak) [reply soc/arm/atmel_sam/same70/dts_fixup.h:130]: just historic, leaving what that part of the define alone.

File: soc/arm/nxp_imx/rt/dts_fixup.h (hunk 2 above)
Comment (galak) [reply soc/arm/nxp_imx/rt/dts_fixup.h:28]: just historic, leaving what that part of the define alone.

**Review by cvinayak — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Comment by anangl** [comment]
@galak A related change in the hal_nordic module is also needed (see https://github.com/zephyrproject-rtos/hal_nordic/pull/32).

**Review by ioannisg — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
needs rebase

File: doc/zephyr.doxyfile.in [hunk 3]
```diff
@@ -1996,7 +1996,8 @@ PREDEFINED             = "CONFIG_ARCH_HAS_CUSTOM_BUSY_WAIT" \
                          "__aligned(x)=" \
                          "__printf_like(x, y)=" \
                          "__attribute__(x)=" \
-                         "__syscall="
+                         "__syscall=" \
+                         "DT_DOXYGEN"
```
Comment (pabigot) [inline doc/zephyr.doxyfile.in:2000]: unrelated, undocumented, change?  What's this for?  (Is this commit from some other PR?  @mbolivar-nordic)

Later: Looks like it's used to generate documentation in Doxygen, but IMO this should not be named to suggest it's restricted to devicetree usage.  There are many other cases where conent that is conditionally included in a header should be included or excluded for Doxygen-generated material.  In other projects I've done this in `doxygen.cfg.in`:
```
# Use "#if defined(@PROJECT_UC@_DOXYGEN)" when providing a stub declaration
# to which documentation is attached.
# Use "/* @cond DOXYGEN_EXCLUDE */ ... /* @endcond */" around the real
# declarations.
PREDEFINED += "@PROJECT_UC@_DOXYGEN"
```

**Review by pabigot — COMMENTED** [review]
This seems to make some sense, in spite of it possibly starting a slide down a slope to a massive `chosen` section.

File: doc/zephyr.doxyfile.in (hunk 3 above)
Comment (galak) [reply doc/zephyr.doxyfile.in:2000]: This was pull in from PR #23891 to get the commits from that PR related to devicetree/zephyr.h

Comment (pabigot) [reply doc/zephyr.doxyfile.in:2000]: OK.  Now I see this:
```
 * Getting doxygen to play along with all the dts-specific ifdeffery
￼ * proved too complex for DT_CHOSEN_ZEPHYR_FLASH_LABEL, so we document
￼ * everything under a DT_DOXYGEN define.
```
in the point of use, so I guess it's explained.  I still think a generic "only emit this when running under Doxygen" solution is more useful, but that's probably more hassle to get through.

**Comment by galak** [comment]
Reworked this PR based on discussion in dev-review.  Pulled in some commits from @mbolivar-nordic PR on FLASH.  This now uses a chosen property to specify the entropy device.

File: include/devicetree/zephyr.h [hunk 4]
```diff
@@ -45,6 +45,8 @@
 #define DT_CHOSEN_ZEPHYR_FLASH_LABEL DT_LABEL(DT_CHOSEN(zephyr_flash))
 #endif
 
+#define DT_CHOSEN_ZEPHYR_ENTROPY_LABEL DT_LABEL(DT_CHOSEN(zephyr_entropy))
```
[the remark this answers is not in the document]
Comment (galak) [reply include/devicetree/zephyr.h:41]: I think DT_NODE_HAS_PROP covers both if the node exists and if the property exists.

[the remark this answers is not in the document]
Comment (galak) [reply include/devicetree/zephyr.h:41]: What part should be under DT_DOXYGEN?

[the remark this answers is not in the document]
Comment (mbolivar-nordic) [reply include/devicetree/zephyr.h:41]: > I think DT_NODE_HAS_PROP covers both if the node exists and if the property exists.

Right you are. OK then.

> What part should be under DT_DOXYGEN?

Never mind, I see it now.

**Review by mbolivar-nordic — APPROVED** [review]

**Review by pabigot — APPROVED** [review]

**Pull request merged**